### DISCUSSION TITLE
Advice on how to prevent detection

### DISCUSSION THREAD

**Discussion by hagak** [blakeblackshear/frigate#3028] [Ask A Question]
I have been trying to prevent Frigate from detecting my parked truck when lighting changes trigger motion.  I put a motion mask over the area where I park the truck however it still will trigger.

Attached is an image showing the most recent trigger that happened.  NOTE the truck has been parked there for a few days now so it is not the case that Frigate tracked the truck into the motion masked area.

Do I need to make the mask larger?
![Screenshot 2022-03-29 101953](https://user-images.githubusercontent.com/9702199/160633528-eba36be3-0580-4ec1-a253-0cae24841710.png)
![Screenshot 2022-03-29 101933](https://user-images.githubusercontent.com/9702199/160633540-b2535369-a018-4741-87f0-13c94d53b742.png)

**Comment by hagak** [comment]
Ok so I extended the motion mask to be larger, i even put in an object mask for car in that same area.  STILL get a car detected in that location.  why?  How can I tell Frigate to NEVER detect anything there.

**Comment by NickM-27** [comment]
Like [the docs say](https://docs.frigate.video/configuration/masks) motion masks are not to be used for blocking out objects. Motion anywhere in an area around the truck will cause it to be detected. Motion masks should be used to block out trees, timestamps, etc. areas of motion that won't be from objects of interest.

As far as what to do, there are a few options:

1. **If you still want to track the truck but not have events while it is parked:**
Setup zones. Put one zone where your truck enters `driveway` and add this to `required_zones` for events and snapshots.
NOTE: If this route is taken then other zones will need to be setup for people, animals, etc. to have events be created.
2. **If you just don't want the truck tracked at all while parked:**
Use an object mask and put it in the area around the bottom center of the bounding box. This will keep it from being detected while parked there.

**Reply by NickM-27** [reply]
I made a doodle for the second option:
![160633528-eba36be3-0580-4ec1-a253-0cae24841710](https://user-images.githubusercontent.com/14866235/160645995-b4b966f6-4797-45fd-9222-232fa47bcc2b.png)

**Reply by NickM-27** [reply]
blue box is where you would want the car mask to be.

**Reply by hagak** [reply]
Thanks I need to look into required_zones.

Question I have an object mask for my car covering the same area as the motion mask and yet it still triggers. 

I also must not understand how Frigate looks for objects, I assumed when it detect motion it would look for object within the area of the motion.  But it sounds like the way it works is motion just triggers it to look for objects everywhere.   Is this correct?

**Reply by NickM-27** [reply]
> Question I have an object mask for my car covering the same area as the motion mask and yet it still triggers.

If it is an object mask as in the first picture then that wouldn't work because the bottom center of the bounding box is not within that object mask. 

> I also must not understand how Frigate looks for objects, I assumed when it detect motion it would look for object within the area of the motion. But it sounds like the way it works is motion just triggers it to look for objects everywhere. Is this correct?

Your initial thought was correct, frigate does intelligently use motion to only look in that area. However, it doesn't work in the abnormal polygon shapes like the mask was drawn. It works in squares, so even if motion didn't occur within the mask you made there could still have been motion right outside of it and then the truck fits inside of the 320x320 region used to look for objects.

**Comment by hagak** [comment]
Increasing the size of the object mask seems to have solved that issue.  Thanks for the assistance.
